src/qilisdk/analog/schedule.py

```diff
@@ -145,25 +147,23 @@ def coefficients(self) -> dict[str, Interpolator]:
     @property
     def T(self) -> float:
         """Total annealing time of the schedule."""
-        return max(self.tlist)
+        if self._max_time is not None:
+            return self._get_value(self._max_time)
+        max_t = self._get_coefficients_max_time()
+        return self._get_value(max_t)

```

Comment:
Should this also cache `max_t`? Or is `self._max_time` reserved for a user-specified explicit max time? 

```suggestion
        self._max_time = self._get_coefficients_max_time()
        return self._get_value(self._max_time)
```

Reply:
no in this case we shouldn't update the self._max_time because we want it to evaluate at each iteration. So basically the self._max_time will force the schedule to be of that time, whereas in this case we want to check what is the maximum time in the schedule.